services/admin-config/src/test/kotlin/AdminConfigServiceTest.kt

```diff
@@ -0,0 +1,260 @@
+            val service = createServiceWithConfig(config)
+
+            val ruleSet = service.loadAdminConfig(context, ORGANIZATION_ID).getRuleSet(null)
+
+            ruleSet.copyrightGarbageFile shouldBe "testCopyrightGarbageFile"
+            ruleSet.licenseClassificationsFile shouldBe "testLicenseClassificationsFile"
+            ruleSet.resolutionsFile shouldBe "testResolutionsFile"
+            ruleSet.evaluatorRules shouldBe "testEvaluatorRules"
+        }
+
+        "return a named rule set" {
+            val config = """
+                    ruleSets {
+                      customRuleSet1 {
+                        copyrightGarbageFile = "testCopyrightGarbageFile1"
+                        licenseClassificationsFile = "testLicenseClassificationsFile1"
+                        resolutionsFile = "testResolutionsFile1"
+                        evaluatorRules = "testEvaluatorRules1"
+                      }  
+                      customRuleSet2 {
+                        copyrightGarbageFile = "testCopyrightGarbageFile2"
+                        licenseClassificationsFile = "testLicenseClassificationsFile2"
+                        resolutionsFile = "testResolutionsFile2"
+                        evaluatorRules = "testEvaluatorRules2"
+                      }  
+                    }
+                """.trimIndent()
+            val service = createServiceWithConfig(config)
+
+            val ruleSet = service.loadAdminConfig(context, ORGANIZATION_ID).getRuleSet("customRuleSet1")
+
+            ruleSet.copyrightGarbageFile shouldBe "testCopyrightGarbageFile1"
+            ruleSet.licenseClassificationsFile shouldBe "testLicenseClassificationsFile1"
+            ruleSet.resolutionsFile shouldBe "testResolutionsFile1"
+            ruleSet.evaluatorRules shouldBe "testEvaluatorRules1"
+        }
+
+        "throw an exception for an unknown rule set name" {
+            val config = """
+                    ruleSets {
+                      someRuleSet {
+                        copyrightGarbageFile = "testCopyrightGarbageFile1"
+                        licenseClassificationsFile = "testLicenseClassificationsFile1"
+                        resolutionsFile = "testResolutionsFile1"
+                        evaluatorRules = "testEvaluatorRules1"
+                      }  
+                    }
+                """.trimIndent()
+            val service = createServiceWithConfig(config)
+
+            val nonExistentRuleSetName = "nonExistentRuleSet"
+            val exception = shouldThrow<NoSuchElementException> {
+                service.loadAdminConfig(context, ORGANIZATION_ID).getRuleSet(nonExistentRuleSetName)
+            }
+
+            exception.message shouldContain nonExistentRuleSetName
+            exception.message shouldContain "No rule set"
+        }
+
+        "set undefined properties in rule sets from the default rule set" {
+            val config = """
+                    defaultRuleSet {
```

Comment:
For the user it might be nicer to have this as a rule set with the name "default" in the list below, and to treat it like any other rule set just with a reserved name, what do you think?

Reply:
The default rule set is special, as its properties are inherited by other rule sets unless they override them. I thought this would be more obvious if it was declared in a more prominent place in the configuration. If you have a longer list of rule sets and the default is just one of them, it is hard to see what are the default values.

Reply:
I have no clear preference, it just came to my mind during the review.